### PR TITLE
Changelog: request body/props changed

### PR DESCRIPTION
**Pull request by blva**
[request prop/body changed: nullable field changed](https://github.com/Tufin/oasdiff/pull/346/commits/c69f82d1e2fc94107ef34c27700f7aeb50e32ee3)

[request prop changed: required set to true/false](https://github.com/Tufin/oasdiff/pull/346/commits/c91e5656466b707b65535bb68f0db240c6f99ff7)

[request props changed: added/removed required/optional property (note - merging 2 checkers)](https://github.com/Tufin/oasdiff/pull/346/commits/629c620e0d39ddef870a22264eff2bea6c2b93c8)

[request body/props changed: default value](https://github.com/Tufin/oasdiff/pull/346/commits/394f63a6cc8cc0e4fc18f509d4162e549b0d6e23)

[request props: writeonly/readonly](https://github.com/Tufin/oasdiff/pull/346/commits/f264ad5dd928f1c31c2005682313d98cb8362f84)

[request props/body: schema type/format changed](https://github.com/Tufin/oasdiff/pull/346/commits/9d8027652640e6b4409fdd62890e30a593adc11c)

### CODE REVIEW

**Review by ciprian-tibulca — APPROVED** [review]
LGTM

File: checker/check-request-property-became-not-nuallable.go [hunk 1]
```diff
@@ -34,12 +36,23 @@ func RequestPropertyBecameNotNullableCheck(diffReport *diff.Diff, operationsSour
...
+					result = append(result, ApiChange{
+						Id:          requestBodyBecameNullableId,
+						Level:       INFO,
+						Text:        config.i18n(requestBodyBecameNullableId),
+						Operation:   operation,
+						Path:        path,
+						Source:      source,
+						OperationId: operationItem.Revision.OperationID,
 					})
 				}
```
Comment (reuvenharrison) [inline checker/check-request-property-became-not-nuallable.go]: perhaps add ```if mediaTypeDiff.SchemaDiff.NullableDiff != nil``` around this block and avoid testing it twice

File: checker/check-request-property-default-value-changed.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+						OperationId: operationItem.Revision.OperationID,
+						Path:        path,
+						Source:      source,
+					})
+				}
+
+				CheckModifiedPropertiesDiff(
+					mediaTypeDiff.SchemaDiff,
+					func(propertyPath string, propertyName string, propertyDiff *diff.SchemaDiff, parent *diff.SchemaDiff) {
+						if propertyDiff == nil || propertyDiff.Revision == nil || propertyDiff.Revision.Value == nil || propertyDiff.DefaultDiff == nil {
```
Comment (reuvenharrison) [inline checker/check-request-property-default-value-changed.go]: 1. propertyDiff.Revision.Value is not used below, so no need to check it
2. other checkers assume that Base and Revision fields are not nil, I think we can do the same here (for consistency)

Comment (blva) [reply checker/check-request-property-default-value-changed.go]: this was copied from check-response-property check which had a fix related to a nil pointer https://github.com/Tufin/oasdiff/pull/272, I think we should keep it

Comment (reuvenharrison) [reply checker/check-request-property-default-value-changed.go]: But in this case you are not using propertyDiff.Revision so what's the point in checking it?

**Review by reuvenharrison — APPROVED** [review]

File: checker/check-request-property-default-value-changed.go (hunk 2 above)
Comment (blva) [reply checker/check-request-property-default-value-changed.go]: ah! you're right, removed! thanks. also, I don't have permissions to merge the PR even with your approval

**Pull request merged**